homeassistant/components/mochad.py

```diff
@@ -39,18 +41,25 @@ def setup(hass, config):
     conf = config[DOMAIN]
     host = conf.get(CONF_HOST)
     port = conf.get(CONF_PORT)
+    conf_mqtt = config[DOMAIN_MQTT]
```

Comment:
This will error if no mqtt section is in the config.

Reply:
If we want to optionally depend on the mqtt component, we have to check that it's set up first. See the mysensors component for an example:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mysensors/gateway.py#L89

Note that that example is async, wo we have to modify it before using it here.

Reply:
Added a synched setup dependency of mqtt